src/mqt/qmap/visualization/visualize_search_graph.py

```diff
@@ -0,0 +1,2116 @@
+        search_node_colorbar_spacing (float): Spacing between multiple colorbars. Defaults to 0.06.
+        search_node_height (str | Callable[[SearchNode], float] | Sequence[str | Callable[[SearchNode], float]]): Function mapping a mqt.qmap.visualization.SearchNode to a float value to be used as z-value in 3D search graphs or a preset data feature (&quot;total_cost&quot; | &quot;fixed_cost&quot; | &quot;heuristic_cost&quot; | &quot;lookahead_penalty&quot;). Or a list any of such functions/data features, to draw multiple points per search node. Defaults to "total_cost".
+        draw_stems (bool): If a vertical stem should be drawn in 3D search graphs to each search node. Defaults to False.
+        stems_width (float): Width of stems in 3D search graphs. Defaults to 0.7.
+        stems_color (str): Color of stems in 3D search graphs (in CSS format). Defaults to "#444".
+        stems_dash (str): Dashing of stems in 3D search graphs (in CSS format). Defaults to "solid".
+        show_search_progression (bool): If the search progression should be animated. Defaults to True.
+        search_progression_step (int): Step size (in number of nodes added) of search progression animation. Defaults to 10.
+        search_progression_speed (float): Speed of the search progression animation. Defaults to 2.
+        plotly_settings (MutableMapping[str, MutableMapping[str, any]] | None): Direct plotly configuration dictionaries to be passed through. Defaults to None.
+        ```
+        {
+            "layout": settings for plotly.graph_objects.Layout (of subplots figure)
+            "arrows": settings for plotly.graph_objects.layout.Annotation
+            "stats_legend": settings for plotly.graph_objects.layout.Annotation
+            "search_nodes": settings for plotly.graph_objects.Scatter resp. ...Scatter3d
+            "search_edges": settings for plotly.graph_objects.Scatter resp. ...Scatter3d
+            "architecture_nodes": settings for plotly.graph_objects.Scatter
+            "architecture_edges": settings for plotly.graph_objects.Scatter
+            "architecture_edge_labels": settings for plotly.graph_objects.Scatter
+            "search_xaxis": settings for plotly.graph_objects.layout.XAxis resp. ...layout.scene.XAxis
+            "search_yaxis": settings for plotly.graph_objects.layout.YAxis resp. ...layout.scene.YAxis
+            "search_zaxis": settings for plotly.graph_objects.layout.scene.ZAxis
+            "architecture_xaxis": settings for plotly.graph_objects.layout.XAxis
+            "architecture_yaxis": settings for plotly.graph_objects.layout.YAxis
+        }
+        ```
+
+    Raises:
+        TypeError: If any of the arguments are invalid.
+
+    Returns:
+        Widget: An interactive IPython widget to visualize the search graph.
+    """
+    # check and process all parameters
+    if plotly_settings is None:
```

Comment:
The options here show quite some duplication with the options in the other visualisation file. Do you see any way to encapsulate these into a centralized place so that this does not become a burden to maintain?

Reply:
Well, there is always the option of encapsulating such settings in a dedicated settings object (or TypedDict). Though this always adds some complexity for the user, as the settings are then not directly exposed by the function itself (especially IDE support and Intelisense always take a hit in my experience).

We already have a very similar situation with `compile` and the `Configuration` class (where the parameters of `compile` are basically a duplication of the members of `Configuration` and are mostly just copied into an object of that type). I decided to keep the visualization module consistent with that design choice. However, since I already experienced maintenance problems multiple times with exactly that duplication between `compile` and `Configuration`, I'm also very open to go the TypedDict route.

Reply:
I see your point. It is really unfortunate that IDE support gets "lost" so easily when trying to write cleaner code in these circumstances.
Over in QCEC we tried to go the route with the typed dict (see https://github.com/cda-tum/mqt-qcec/blob/main/src/mqt/qcec/verify.py). While that completely removed the code duplication, IDE support is still lacking.
In the end, staying consistent is probably more important at first. So maybe let's keep it the way it is for now (also to get this PR merged sooner than later) and rethink our design choices as a follow-up. Does that make sense?